py3status/modules/xrandr_toggle.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Toggle monitor output from the status bar using xrandr.
+
+Configuration parameters:
+    cache_timeout: how often we refresh usage in seconds (default: 10s)
+    distext: custom text or symbol for disconnected status
```

Comment:
```
                self.mon = next((x for x in monitors                    
                                if x['id'] == self.source),
                                None)
```

yeah that looks nasty.

some thoughts

a) create `_get_monitor()` that encapsulates it.

b) for the named monitors create a dict `monitor[name] = ...` then easy to access no need to go through the loop.  I'm not sure how the numbers work.  Do they always map to a screen name?  Anyway you could use an orderedDict and then get by index `list(d.items())[i]`